lib/attrsets.nix

```diff
@@ -826,6 +826,48 @@ rec {
       n: a: foldr (name: o: o // { ${name} = op n.${name} (a.${name} or nul); }) a (attrNames n)
     ) { } list_of_attrs;
 
+  /**
+    Apply fold function to values grouped by key. Like
+    [`lib.attrsets.foldAttrs`](#function-library-lib.attrsets.foldAttrs), but
+    passes an attribute name as a first argument to `op`.
+
+    # Inputs
+
+    `op`
+
+    : A function, given an attribute name, a value and a collector combines the later two.
+
+    `nul`
+
+    : The starting value.
+
+    `list_of_attrs`
+
+    : A list of attribute sets to fold together by key.
+
+    # Type
+
+    ```
+    foldAttrs' :: (String -> Any -> Any -> Any) -> Any -> [AttrSets] -> Any
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.attrsets.foldAttrs'` usage example
+
+    ```nix
+    foldAttrs' (name: item: acc: ["${name}${toString item}"] ++ acc) [] [{ a = 2; b = 3; } { a = 4; }]
```

Comment:
This example is equivalent to the existing

```nix
    zipAttrsWith (name: values: map (v: name + toString v) values) [{ a = 2; b = 3; } { a = 4; }]
```

Only missing feature would be the accumulator. But the mentioned use-cases didn't really convince the need for that.